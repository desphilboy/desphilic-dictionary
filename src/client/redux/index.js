import { createStore, compose, combineReducers, applyMiddleware } from 'redux';
import { reducer as form } from 'redux-form'
import { lookUpReducer as lookUp } from './look-up/look-up.reducer.js';
import 'core-js';
import 'regenerator-runtime/runtime';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './redux-sagas/root-saga';

const initialState = {
	lookUp: {
		inProgress: false,
		data: false,
		error: false,
	},
};

/* eslint-disable-next-line no-underscore-dangle */
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const sagaMiddleware = createSagaMiddleware();

const reducer = combineReducers({
	lookUp,
	form,
});

export const store = createStore(reducer, initialState, composeEnhancers(applyMiddleware(sagaMiddleware)));

sagaMiddleware.run(rootSaga);