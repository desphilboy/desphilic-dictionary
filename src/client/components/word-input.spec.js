
import React from 'react';
import { shallow, mount } from 'enzyme';
import Enhanced, * as wordInputLogix from './word-input';
import configureMockStore from 'redux-mock-store';
import * as Redux from 'redux';
import { Provider } from 'react-redux'; 
import * as ReduxForm from 'redux-form';


describe('component word input', () => {
	describe('the UI View', () => {
		it('matches the snapshot', () => {
			const wrapper = shallow(<wordInputLogix.WordInput handleSubmit={jest.fn()}/>);

			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('mapDispatchToProps', () => {
		const dispatch = jest.fn();
		let map;

		beforeEach(() => {
			map = wordInputLogix.mapDispatchToProps(dispatch);
		});

		it('has an onSubmit', () => {
			expect(map.onSubmit).toEqual(expect.any(Function));
		});

		it('onSubmit calls dispatch with lookup action', () => {
			map.onSubmit({word: 'someTestWord'});

			expect(dispatch).toHaveBeenCalledWith({
			    payload: {
			        data: false,
			        error: false,
			        inProgress: true,
			        word: 'someTestWord',
			    },
			    type: 'LOOKUP_START',
			});
		});
	});

	describe('enhanced component', () => {
		it('calls compose with connect and reduxForm on the ui component', () => {
			const mockStore = configureMockStore()({}); 
			Redux.compose = jest.fn((C) => (props => <div {...props} /> ));
			ReduxForm.reduxForm = jest.fn(() => 'maloo');

			const wrapper = mount(<Provider store={mockStore} ><Enhanced p="p"/></Provider>);
			//expect(ReactRedux.connect).toHaveBeenCalledWith('maloo');
			expect(Redux.compose).toHaveBeenCalledWith('paloo', 'maloo');
		});
	})
});
