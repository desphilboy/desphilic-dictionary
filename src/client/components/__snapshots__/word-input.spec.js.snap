// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component word input the UI View matches the snapshot 1`] = `ShallowWrapper {}`;
